Python/Product/PythonTools/PythonTools/Debugger/DebugLaunchHelper.cs

```diff
@@ -84,40 +85,82 @@ public static class DebugLaunchHelper {
             return SubstitutionPattern.Replace(
                 str,
                 m => {
-                    string value;
-                    return environment.TryGetValue(m.Groups[1].Value, out value) ? value : "";
+                    return environment.TryGetValue(m.Groups[1].Value, out string value) ? value : "";
                 }
             );
         }
 
+        private static string GetArgs(LaunchConfiguration config) {
+            var args = string.Join(" ", new[] {
+                    config.InterpreterArguments,
+                    config.ScriptName == null ? "" : ProcessOutput.QuoteSingleArgument(config.ScriptName),
+                    config.ScriptArguments
+                }.Where(s => !string.IsNullOrEmpty(s)));
+
+            if (config.Environment != null) {
+                args = DoSubstitutions(config.Environment, args);
+            }
+            return args;
+        }
+
+        private static string GetOptions(IServiceProvider provider, LaunchConfiguration config) {
+            var pyService = provider.GetPythonToolsService();
+            return string.Join(";", GetGlobalDebuggerOptions(pyService)
+                .Concat(GetLaunchConfigurationOptions(config))
+                .Where(s => !string.IsNullOrEmpty(s))
+                .Select(s => s.Replace(";", ";;"))
+            );
+        }
+
+        private static string GetLaunchJsonForVsCodeDebugAdapter(IServiceProvider provider, LaunchConfiguration config) {
+            bool nativeDebug = config.GetLaunchOption(PythonConstants.EnableNativeCodeDebugging).IsTrue();
```

Comment:
This should never be true on this code path. Basically, if native debugging is enabled, we should be using the old engine (which in that case just bails out and lets Concord handle it).